### PR TITLE
updates.py refactor

### PR DESCRIPTION
**Pull request by Josca**
Refactored webapp/updates.py to be easier to read.

### CODE REVIEW

File: webapp/test/test_updates.py [hunk 1]
```diff
@@ -62,49 +62,62 @@ def setup_api(self, load_cache):
...
-        response = {}
-        repo_ids = self.updates_api._process_repositories(UPDATES_JSON, response)
-        assert len(repo_ids) == 1
+        repositories_list, repo_ids = self.updates_api._get_repository_list(UPDATES_JSON)
+        releasever, repo_ids = self.updates_api._get_releasever(UPDATES_JSON, repo_ids)
+        basearch, repo_ids = self.updates_api._get_basearch(UPDATES_JSON, repo_ids)
+        assert repositories_list == ['rhel-7-server-rpms']
+        assert releasever == '7Server'
+        assert basearch == 'x86_64'
+        assert repo_ids == {1}
```
Comment (MichaelMraka) [inline webapp/test/test_updates.py:71]: it no more tests self.updates_api._process_repositories() but 3 different methods

Comment (Josca) [reply webapp/test/test_updates.py:71]: Well, these methods depend on each other and relate to repository_list processing so I kept them together. Should I split it to separate tests or just rename current test?

Comment (Josca) [reply webapp/test/test_updates.py]: Ok, I've created a separate tests for the new functions. It will be better.

Comment (MichaelMraka) [reply webapp/test/test_updates.py]: that's what I mean; keep test for old method and create one for new

**Review by MichaelMraka — APPROVED** [review]

**Review by jdobes — APPROVED** [review]

**Pull request merged**